### PR TITLE
Polar contour

### PR DESCRIPTION
**Pull request by brianpm**
This addresses https://github.com/NCAR/ADF/issues/101

One script to go into `scripts/plotting` called `polar_map.py`.

Functions:
- `polar_map(adfobj)` : the main function that gets information from ADF. Closely follows the global maps.
- `make_polar_plot` : the function that constructs the plot, needs the data fields and the hemisphere specified.
- `domain_stats` : a little helper function to get the mean/min/max values in the domain being plotted (for annotation)

I also added a function to `lib/plotting_functions.py` called `get_difference_colors(values)`. This returns a Matplotlib normalization and colormap. It's basically stripped from lower down in the plotting functions, so other functions can use it. 

One other change is a typo correction in the taylor diagram that I encountered while testing.

### CODE REVIEW

File: lib/plotting_functions.py [hunk 1]
```diff
@@ -36,6 +36,33 @@ def use_this_norm():
...
+    # color normalization for difference
+    if ((dmin < 0) and (0 < dmax)):
+        dnorm = normfunc(vmin=np.min(values), vmax=np.max(values), vcenter=0.0)
+        cmap = mpl.cm.RdBu_r
+    else:
+        dnorm = mpl.colors.Normalize(vmin=np.min(values), vmax=np.max(values))
+        if dmin > 0:
+            cmap = mpl.cm.OrRd
+        elif dmax < 0:
+            cmap = mpl.cm.BuPu_r
```
Comment (nusbaume) [inline lib/plotting_functions.py:60]: I believe these should be modified like so:

```suggestion
        if dmin >= 0:
            cmap = mpl.cm.OrRd
        elif dmax <= 0:
            cmap = mpl.cm.BuPu_r
```

That way we can capture situations where the limits are exactly zero:

File: scripts/plotting/polar_map.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+import cartopy.feature
+from cartopy.util import add_cyclic_point
+
+# ADF library
+import plotting_functions as pf
+
+
+def polar_map(adfobj):
+    """
+    This script/function is generates polar maps of model fields with continental overlays.
```
Comment (nusbaume) [inline scripts/plotting/polar_map.py:23]: Typo here (`function is generates` -> `function generates`)

File: scripts/plotting/polar_map.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+    x_region_mean = x_region.weighted(np.cos(x_region['lat'])).mean().item()
+    x_region_min = x_region.min().item()
+    x_region_max = x_region.max().item()
+    return x_region_mean, x_region_max, x_region_min
+
+
+
+def make_polar_plot(d1:xr.DataArray, d2:xr.DataArray, difference:Optional[xr.DataArray]=None, domain:Optional[list]=None, hemisphere:Optional[str]=None, **kwargs):
+    '''
+    Make a stereographic polar polt for the given data and hemisphere.
```
Comment (nusbaume) [inline scripts/plotting/polar_map.py:239]: Typo here (`polt` -> `plot`).

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good!  Just some minor requests.

File: lib/plotting_functions.py (hunk 1 above)
Comment (brianpm) [reply lib/plotting_functions.py:60]: Yeah, I agree. Done.

File: scripts/plotting/polar_map.py (hunk 2 above)
Comment (brianpm) [reply scripts/plotting/polar_map.py:23]: Done.

File: scripts/plotting/polar_map.py [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+                    #End if (dimensions check)
+
+                else: #odata dimensions check
+                     print("\t \u231B skipping polar map for {} as it doesn't have only lat/lon dims.".format(var))
+
+                #End if (dimensions check)
+            #End for (case loop)
+        #End for (obs/baseline loop)
+    #End for (variable loop)
+
```
[the remark this answers is not in the document]
Comment (brianpm) [reply scripts/plotting/polar_map.py:221]: Added plt.close() for both figures right after they are saved.

**Comment by nusbaume** [comment]
@brianpm can you double-check that you committed and pushed the latest changes to your "polar_contour" branch?  I am unable to find any new commits or code changes.  Thanks!

**Comment by brianpm** [comment]
Strange, I know I pushed it before. I just did it again (maybe I misspelled the branch name or something). See if you can see the changes now.

**Comment by nusbaume** [comment]
@brianpm I can see the changes now, thanks!  

Also, I pushed a couple changes to add the polar plots to the website and default config file, so this should be good to go now.  

Thanks again for developing the plot!

**Review by nusbaume — APPROVED** [review]

**Pull request merged**